src/main/java/seedu/bigpp/command/buildercommand/BuilderSelectCommand.java

```diff
@@ -65,11 +67,20 @@ public String executeCommand(DataStorage dataStorage) throws BuilderMissingSelec
 
         int pcIndex = UI.builderMenu.getPCIndex();
 
+        float currentCost = dataStorage.pcList.get(pcIndex).getCost();
+        int pcBudget = dataStorage.pcList.get(pcIndex).getBudget();
+        float componentPrice = dataStorage.stringToComponentListMap.get(componentTypeString).get(componentIndex)
+                .getPrice();
+
+        if (componentPrice + currentCost > pcBudget) {
+            throw new BuilderExceedBudgetException();
+        }
+
```

Comment:
This wont work for inifinite budget i believe

Reply:
how about i check for budget != -1 and then do this